app/handlers/newflow/verify_email_by_pin.rb

```diff
@@ -0,0 +1,34 @@
+module Newflow
+    # Marks an `EmailAddress` as `verified` if it matches the passed-in `EmailAddress`'s pin
+    # and then marks the owner of the email address as 'activated'.
+    class VerifyEmailByPin
+    lev_handler
+    uses_routine ConfirmByPin
+    uses_routine ActivateUser
+
+    paramify :confirm do
+      attribute :pin, type: String
+      validates :pin, presence: true
+    end
+
+    def authorized?
+      true
+    end
+
+    def handle
+      result = ConfirmByPin.call(contact_info: options[:email_address], pin: confirm_params.pin)
+      if result.errors.any?
+        fatal_error(
+          code: :invalid_confirmation_pin,
+          offending_inputs: [:pin],
+          message: I18n.t(:"login_signup_form.invalid_confirmation_pin")
+        )
+      end
```

Comment:
Good catch. I need will Dante's help on this one because I have not been able to call `ConfirmByPin` any other way than by the **not recommended** way: `ConfirmByPin.call`. So in order to sort of "transfer_errors_from" that Handler to mine I had to do that... check `errors.any?`